### ISSUE TITLE
Installation / NPM warnings

### ISSUE THREAD

**Issue by D4V3M0NK** [Ylianst/MeshCentral#2383]
Installing MC2v0.7.85:
```
npm WARN deprecated xmldom@0.1.31: Deprecated due to CVE-2021-21366 resolved in 0.5.0
npm WARN deprecated debug@3.2.6: Debug versions >=3.2.0 <3.2.7 || >=4 <4.3.1 have a low-severity ReDos regression when used in a Node.js environment. It is recommended you upgrade to 3.2.7 or 4.3.1. (https://github.com/visionmedia/debug/issues/797)
```
I've typically kinda ignored the NPM warnings upon install of MC2, but considering one of them relates to a CVE, it peaked my interest more than usual. Immediately after the install:
```
$ cd MESH_INSTALL_FOLDER
$ npm list | grep -E 'xmldom|debug'
│ │ ├─┬ debug@2.6.9
│ │ ├─┬ debug@4.3.1
│ │ ├─┬ debug@3.2.6
│ ├── xmldom@0.1.31
```
I am (by far) a complete n00b when it comes to node, but through searching on how to upgrade node modules, I assumed that using `npm upgrade xmldom debug` (within the MC2 install folder), or `npm upgrade -g xmldom debug` elsewhere would update these two modules. However when I run that, nothing really happens, and I check afterwards, the versions are still the same.

Is this something we can even control, or is it like PHP's composer, inasmuch there's a master file containing the versions of the modules that can only be updated by the MC2 team?

**Comment by D4V3M0NK** [comment]
```
cd MESH_INSTALL_FOLDER/node_modules/xmldom
npm i xmldom@latest
WARN engine xmldom@0.5.0: wanted: {"node":">=10.0.0"} (current: {"node":"8.10.0","npm":"3.5.2"})
npm WARN deprecated xmldom@0.1.31: Deprecated due to CVE-2021-21366 resolved in 0.5.0
/MESH_INSTALL_FOLDER
├── archiver@4.0.2 
├─┬ meshcentral@0.7.85 
│ └── xmldom@0.1.31 
├── mysql@2.18.1 
├── otplib@10.2.3 
└── xmldom@0.5.0
```
Worked! However, when attempting to do the same with debug:
```
cd MESH_INSTALL_FOLDER/node_modules/debug
$ npm i debug@latest
npm ERR! Linux 4.15.0-136-generic
npm ERR! argv "/usr/bin/node" "/usr/bin/npm" "i" "debug@4.3.1"
npm ERR! node v8.10.0
npm ERR! npm  v3.5.2
npm ERR! code ENOSELF

npm ERR! Refusing to install debug as a dependency of itself
npm ERR! 
npm ERR! If you need help, you may report this error at:
npm ERR!     <https://github.com/npm/npm/issues>

npm ERR! Please include the following file with any support request:
npm ERR!     MESH_INSTALL_FOLDER/node_modules/debug/npm-debug.log
```

Should I even be monkeying around with this? I appreciate that the MC2 team tests against particular versions...

**Comment by Ylianst** [comment]
Already [fixed on this commit](https://github.com/Ylianst/MeshCentral/commit/a700c9afb032ca3db3cecbca7beacd5931236c0b). Will be in MeshCentral v0.7.86.
